### PR TITLE
Migrate probabilisticsamplerprocessor from Collector core to Collector contrib

### PR DESCRIPTION
**Pull request by Saber-W**
**Description:**
This PR migrates the probabilisticsamplerprocessor component from the Collector core repository to the Collector contrib repository.

This PR retains the commit history from core and includes plumbing fixes for go.mod, makefile, components.go.

**Link to tracking Issue:**
[opentelemetry-collector/#3474](https://github.com/open-telemetry/opentelemetry-collector/issues/3474)
[opentelemetry-collector/#3645](https://github.com/open-telemetry/opentelemetry-collector/issues/3645)

### CODE REVIEW

**Comment by alolita** [comment]
Hi @bogdandrutu @tigrannajaryan based on our discussion last week, we're starting to file PRs to move each core component to contrib. Please review and merge. Thanks!

File: processor/probabilisticsamplerprocessor/probabilisticsampler_test.go [hunk 1]
```diff
@@ -22,11 +22,11 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/idutils"
```
Comment (bogdandrutu) [inline processor/probabilisticsamplerprocessor/probabilisticsampler_test.go:25]: The order is not good since this is local and collector is other

Comment (bogdandrutu) [reply processor/probabilisticsamplerprocessor/probabilisticsampler_test.go:25]: I think goimports has that limitation. Just merge all the blocks and run again the goimports and will work

File: processor/probabilisticsamplerprocessor/go.mod [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor
+
+go 1.16
+
+require (
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-20210820200022-94ff24b8ebbd
```
Comment (bogdandrutu) [inline processor/probabilisticsamplerprocessor/go.mod:6]: Use version 0.0.0-0000 or whatever we do in other scenarios and a replace statement for the module to get the local

File: processor/probabilisticsamplerprocessor/Makefile [hunk 3]
```diff
@@ -0,0 +1 @@
+include ../../Makefile.Common
```
Comment (bogdandrutu) [inline processor/probabilisticsamplerprocessor/Makefile:1]: New line

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
```
diff --git a/processor/probabilisticsamplerprocessor/go.mod b/processor/probabilisticsamplerprocessor/go.mod
index 9969e3a29..390f0a1c6 100644
--- a/processor/probabilisticsamplerprocessor/go.mod
+++ b/processor/probabilisticsamplerprocessor/go.mod
@@ -9,4 +9,4 @@ require (
        go.opentelemetry.io/collector/model v0.33.1-0.20210820002854-d3000232f8f6
 )
 
-replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal => ../../internal/coreinternal
\ No newline at end of file
+replace github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal => ../../internal/coreinternal
diff --git a/processor/probabilisticsamplerprocessor/go.sum b/processor/probabilisticsamplerprocessor/go.sum
index 030ff5758..5155d42eb 100644
--- a/processor/probabilisticsamplerprocessor/go.sum
+++ b/processor/probabilisticsamplerprocessor/go.sum
@@ -935,8 +935,6 @@ github.com/onsi/gomega v1.7.0/go.mod h1:ex+gbHU/CVuBBDIJjb2X0qEXbFg53c61hWP/1Cpa
 github.com/onsi/gomega v1.7.1/go.mod h1:XdKZgCCFLUoM/7CFJVPcG8C1xQ1AJ0vpAezJrB7JYyY=
 github.com/onsi/gomega v1.9.0/go.mod h1:Ho0h+IUsWyvy1OpqCwxlQ/21gkhVunqlU8fDGcoTdcA=
 github.com/op/go-logging v0.0.0-20160315200505-970db520ece7/go.mod h1:HzydrMdWErDVzsI23lYNej1Htcns9BCg93Dk0bBINWk=
-github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-20210820200022-94ff24b8ebbd h1:JSdYlAF78Yau6XxyDvrjBnmvbUUZOkTNiicryeMW/ZI=
-github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.0.0-20210820200022-94ff24b8ebbd/go.mod h1:zkt0UqaCY4b5p/ZB31LWHY22EwZTugZYhNsLjh2eUjc=
 github.com/opencontainers/go-digest v1.0.0 h1:apOUWs51W5PlhuyGyz9FCeeBIOUDA/6nW8Oi/yOhh5U=
 github.com/opencontainers/go-digest v1.0.0/go.mod h1:0JzlMkj0TRzQZfJkVvzbP0HBR3IKzErnv2BNG4W4MAM=
 github.com/opencontainers/image-spec v1.0.1 h1:JMemWkRwHx4Zj+fVxWoMCFm/8sYGGrUVojFA6h/TRcI=

```

**Pull request merged**